--- a/include/time.h
+++ b/include/time.h
@@ -26,6 +26,11 @@
 # define _TIME_H	1
 # include <features.h>

+#ifdef __USE_ISOC11
+/* Time base values for timespec_get.  */
+# define TIME_UTC 1
+#endif
+
 __BEGIN_DECLS

 #endif
--- /dev/null
+++ b/librt/timespec_get.c
@@ -0,0 +1,31 @@
+/* Copyright (C) 2011-2024 Free Software Foundation, Inc.
+   This file is part of the GNU C Library.
+
+   The GNU C Library is free software; you can redistribute it and/or
+   modify it under the terms of the GNU Lesser General Public
+   License as published by the Free Software Foundation; either
+   version 2.1 of the License.
+
+   The GNU C Library is distributed in the hope that it will be useful,
+   but WITHOUT ANY WARRANTY; without even the implied warranty of
+   MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+   Lesser General Public License for more details.
+
+   You should have received a copy of the GNU Lesser General Public
+   License along with the GNU C Library; if not, see
+   <https://www.gnu.org/licenses/>.  */
+
+#include <time.h>
+
+
+/* Set TS to calendar time based in time base BASE.  */
+int
+timespec_get (struct timespec *ts, int base)
+{
+  if (base == TIME_UTC)
+    {
+      clock_gettime (CLOCK_REALTIME, ts);
+      return base;
+    }
+  return 0;
+}
--- /dev/null
+++ b/librt/timespec_getres.c
@@ -0,0 +1,32 @@
+/* Get resolution of a time base.
+   Copyright (C) 2021-2024 Free Software Foundation, Inc.
+   This file is part of the GNU C Library.
+
+   The GNU C Library is free software; you can redistribute it and/or
+   modify it under the terms of the GNU Lesser General Public
+   License as published by the Free Software Foundation; either
+   version 2.1 of the License.
+
+   The GNU C Library is distributed in the hope that it will be useful,
+   but WITHOUT ANY WARRANTY; without even the implied warranty of
+   MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+   Lesser General Public License for more details.
+
+   You should have received a copy of the GNU Lesser General Public
+   License along with the GNU C Library; if not, see
+   <https://www.gnu.org/licenses/>.  */
+
+#include <time.h>
+
+
+/* Set TS to resolution of time base BASE.  */
+int
+timespec_getres (struct timespec *ts, int base)
+{
+  if (base == TIME_UTC)
+    {
+      clock_getres (CLOCK_REALTIME, ts);
+      return base;
+    }
+  return 0;
+}
